lib/spack/spack/filesystem_view.py

```diff
@@ -508,6 +629,13 @@ def _check_no_ext_conflicts(self, spec):
 #####################
 # utility functions #
 #####################
+def get_spec_from_file(filename):
+    try:
+        with open(filename, "r") as f:
+            return spack.spec.Spec.from_yaml(f)
+    except IOError:
```

Comment:
You're correct that this was already extant code, but I disagree that it's a misuse of IOError. We want a no-exception method for getting specs from files, so that the calling code can determine whether not getting a spec is an error or not.

Reply:
I actually agree with Peter that this should be more specific -- why does it have to squash the IOError?  Doing that means the calling code knows a lot less about what happened.

In this case, there is actually likely to be something wrong if `from_yaml()` fails, because the calling code got the name from `os.listdir()`, so you know the file should really exist.  If it is corrupt or unreadable, the user likely wants to know, at least with a warning.

Note that `from_yaml()` can take a filename, so you don't need the `open` call here.   would just inline this function and have the calling code determine what is worth catching...